blueflood-core/src/main/java/com/rackspacecloud/blueflood/inputs/formats/AggregatedPayload.java

```diff
@@ -30,28 +39,101 @@
 
+    @NotEmpty
     private String tenantId;
+
+    @EpochRange(maxPast = EpochRangeLimits.BEFORE_CURRENT_TIME_MS,
+            maxFuture = EpochRangeLimits.AFTER_CURRENT_TIME_MS,
+            message = "Out of bounds. Cannot be more than ${maxPast.getValue()} milliseconds into the past. Cannot be more than ${maxPast.getValue()} milliseconds into the future")
     private long timestamp; // millis since epoch.
 
     // this field is optional
+    @org.hibernate.validator.constraints.Range(min=0, max=Long.MAX_VALUE)
     private long flushInterval = 0;
-    
+
+    @Valid
     private BluefloodGauge[] gauges;
+
+    @Valid
     private BluefloodCounter[] counters;
+
+    @Valid
     private BluefloodTimer[] timers;
+
+    @Valid
     private BluefloodSet[] sets;
+
     private BluefloodEnum[] enums;
     
-    private Map<String, Object> metadata;
+    private final List<ErrorResponse.ErrorData> validationErrors;
+
+    private static final ValidatorFactory factory = Validation.buildDefaultValidatorFactory();
+    protected static final Validator validator = factory.getValidator();
+
+    public AggregatedPayload() {
+        this.validationErrors = new java.util.ArrayList<ErrorResponse.ErrorData>();
+    }
+
+    public static AggregatedPayload create(JsonElement json) {
+        AggregatedPayload payload = new Gson().fromJson(json, AggregatedPayload.class);
+        validate(payload);
+
+        return payload;
+    }
 
     public static AggregatedPayload create(String json) {
+
         AggregatedPayload payload = new Gson().fromJson(json, AggregatedPayload.class);
+        validate(payload);
+
         return payload;
     }
-    
+
+    private static void validate(AggregatedPayload payload) {
+        if (payload == null) {
+            throw new InvalidDataException("Invalid request body");
+        }
+
+        Set<ConstraintViolation<AggregatedPayload>> constraintViolations = validator.validate(payload);
+        for (ConstraintViolation<AggregatedPayload> constraintViolation : constraintViolations) {
+            String source = constraintViolation.getPropertyPath().toString().equalsIgnoreCase("valid") ? "" :
+                    constraintViolation.getPropertyPath().toString();
+
+            String metricName = "";
+
+            Object leafBean = constraintViolation.getLeafBean();
+            if (leafBean instanceof BluefloodGauge) {
+                metricName = ((BluefloodGauge) leafBean).getName();
+            } else if (leafBean instanceof BluefloodCounter) {
+                metricName = ((BluefloodCounter) leafBean).getName();
+            } else if (leafBean instanceof BluefloodTimer) {
+                metricName = ((BluefloodTimer) leafBean).getName();
+            } else if (leafBean instanceof BluefloodSet) {
+                metricName = ((BluefloodSet) leafBean).getName();
+            }
```

Comment:
Can we get rid of some of these if statements by making a parent class with the getName() method for these stats?

Should there be an else statement?